apps/homescreen/test/unit/datastore_test.js

```diff
@@ -0,0 +1,370 @@
+      bookmarks.datastore._tasks[0] = {
+        operation: 'remove',
+        id: 0,
+        data: {}
+      };
+      bookmarks.synchronise().then(() => {
+        assert.isTrue(stub.calledOnce);
+        done();
+      });
+    });
+
+    test('should call set() on clear', done => {
+      stub = sinon.stub(bookmarks, 'clear');
+      bookmarks.datastore._tasks[0] = {
+        operation: 'clear',
+        id: 0,
+        data: {}
+      };
+      bookmarks.synchronise().then(() => {
+        assert.isTrue(stub.calledOnce);
+        done();
+      });
+    });
+
+    test('should call updateRevision() when done', done => {
+      stub = sinon.stub(bookmarks, 'updateRevision');
+      bookmarks.datastore._tasks = [{
+        operation: 'done',
+        id: 0,
+        data: null
+      }];
+      bookmarks.synchronise().then(() => {
+        assert.isTrue(stub.calledOnce);
+        done();
+      });
+    });
+  });
+
+  suite('indexedDB backed methods', () => {
+    var objectStores = {};
+    var deletedValues = [];
+    var attachDBTransaction = objectStoreName => {
+      objectStores = {};
+      deletedValues = [];
+      bookmarks.db.transaction = (objectStoreNames, IDBTransactionMode) => {
+        return {
+          objectStore: name => {
+            return {
+              put: value => {
+                if (name !== objectStoreName) {
+                  return;
+                }
+                objectStores[name] = value;
+              },
+              delete: value => {
+                if (name !== objectStoreName) {
+                  return;
+                }
+                deletedValues.push(value);
+              },
+              clear: () => {
+                if (name !== objectStoreName) {
+                  return;
+                }
+              },
+              get: () => {
+                if (name !== objectStoreName) {
+                  return;
+                }
+                return {
+                  set onsuccess(cb) {
+                    cb({
+                      target: {
+                        result: {
+                          id: 'abc/',
+                          data: {}
+                        }
+                      }
+                    });
+                  }
+                };
+              },
+              openCursor: () => {
+                return {
+                  set onsuccess(cb) {
+                    if (objectStoreNames.indexOf(name) === -1) {
+                      return;
+                    }
+
+                    cb({
+                      target: {
+                        result: {
+                          value: {
+                            id: 'abc/',
+                            data: {}
+                          },
+                          continue: () => {
+                          }
+                        }
+                      }
+                    });
+                  }
+                };
```

Comment:
nit, these bracket cascades get very hard to read, I think it'd be worth compressing some of these things to one line. e.g.
cb({ target: {
  result: { value: {
    id: 'abc/',
    data: {}
  },
  continue: () => {}
} });